jetty-ee9/jetty-ee9-webapp/src/main/java/org/eclipse/jetty/ee9/webapp/MetaData.java

```
@@ -757,7 +762,33 @@ public boolean isValidateXml()
      */
     public void setValidateXml(boolean validateXml)
     {
-        _validateXml = validateXml;
+        if (_xmlParser.isValidating() != validateXml)
+        {
+            _xmlParser = new XmlParser(validateXml);
```

I'd rather throw an `IllegalStateException` here as this could silently overwrite the `_xmlParser` containing a previously set `XmlParser` subclass.

---

We can't create the `XmlParser` here. If you look in `WebDescriptor` it uses a subclass of `XmlParser` called `WebDescriptorParser`.

Maybe we keep the new setter for the xmlParser on `MetaData`, changing the signature to `WebDescriptorParser` (and making that class`public` rather than `private`).  Then, change `WebDescriptor` so that if the arg to `parse(xmlParser)` is null, it just uses its static non-validating parser.  That way, if users want to create their own validating parser, they can set it on the `MetaData` instance and it will be used in the calls to `WebDescriptor.parse(xmlParser)`. 

If they just call `MetaData.setValidateXml(true)` instead, we could create a validating `WebDescriptorParser` and use it in calls to `WebDescriptor.parse(xmlParser)`. Although I would probably throw an exception iff there was already a parser instance set on `MetaData`.